### ISSUE TITLE
ssh cp doesn't have the same signature as cp module

### ISSUE THREAD

**Issue by matdrapeau** [saltstack/salt#22751]
The ssh cp wrapper cannot support the same arguments as the cp module
Example:
```
salt-ssh \* cp.get_file /tmp/foo /tmp/bar makedirs=True
machine1:
    TypeError encountered executing cp.get_file: get_file() got an unexpected keyword argument 'makedirs'
```
same for ```gzip``` and ```template```

**Comment by rallytime** [comment]
@matdrapeau Can you post the output of `salt-ssh --versions-report`, please?

**Comment by matdrapeau** [comment]
@rallytime basically any tips of latest branches
You can compare it looking at the function arguments from here:
https://github.com/saltstack/salt/blob/2015.2/salt/modules/cp.py#L122
to here:
https://github.com/saltstack/salt/blob/2015.2/salt/client/ssh/wrapper/cp.py#L11

**Comment by rallytime** [comment]
Thanks for clarifying. 

ping @basepi

**Comment by basepi** [comment]
Thanks, we'll get this fixed.

**Comment by basepi** [comment]
Fixed in #22866
